examples/k8s/hello-kubecon/hello-kubecon.tf

```diff
@@ -0,0 +1,72 @@
+# This is an example of how to deploy the hello-kubecon charm
+# with its own nginx ingress. 
+# Check https://charmhub.io/hello-kubecon for more details.
+#
+# This example deploys both charms and integrates them.
+# Observe how the nginx charm is configured to point to the
+# hello-kubecon internal service and sets an external
+# hostname.
+#
+# After deployed, check that everything is running with
+# a request using the corresponding host header:
+# `curl -H "Host: service.foo" 127.0.0.1`
+# If you'd rather want to see the corresponding page in your
+# browser you can run:
+# echo "127.0.1.1 service.foo" | sudo tee -a /etc/hosts
+# and then visit http://service.foo
+
+
+terraform {
+  required_providers {
+    juju = {
+      source  = "juju/juju"
+      version = "0.4.0"
+    }
+  }
+}
+
+provider "juju" {}
+
+resource "juju_model" "app" {
+  name = "hello-kubecon"
+  config = {
+    logging-config = "<root>=INFO;unit=DEBUG"
+  }
+}
+
+resource "juju_application" "hello_kubecon" {
+  name  = "hello-kubecon"
+  model = juju_model.app.name
+  charm {
+    name = "hello-kubecon"
+  }
+}
+
+resource "juju_application" "nginx" {
+  name  = "ingress"
+  model = juju_model.app.name
+  charm {
+    name = "nginx-ingress-integrator"
+  }
+  config = {
+    service-name     = juju_application.hello_kubecon.name
+    service-port     = "80"
+    service-hostname = "service.foo"
```

Comment:
You don't need to pass these configuration options. These will be handled via the relation. Without any config options at all the charm will be available on the `hello-kubecon` hostname, or you can use an `external-hostname` option to the hello-kubecon charm to set the hostname you want (e.g. `service-foo`) and that will be passed over the relation to nginx-ingress-integrator.